src/dev/tslint/run_tslint_cli.ts

```diff
@@ -70,10 +70,12 @@ export function runTslintCli() {
   );
 
   list.run().catch((error: any) => {
+    process.exitCode = 1;
+
     if (!error.errors) {
       log.error('Unhandled execption!');
       log.error(error);
-      process.exit(1);
+      process.exit();
```

Comment:
Can you explain the reason for this way?

Reply:
Yeah, setting `process.exitCode` tells node that when it naturally exits to use that exit code. I could also just call `process.exit(1);` in both branches of the `.catch()` handler, but I personally prefer to use the natural exit method as it's a little less brute-force and doesn't hide things like child processes or file descriptors that are left open and preventing the process from naturally exiting.

Reply:
That said, calling `process.exit()` in the case that an unhandled exception is caught is important, because an unhandled exception could mean that we hit a failure before we were able to properly clean up. But in the case below, were we are iterating `error.errors`, all cleanup should have run so this sorta relies on that and will fail to exit if that changes in the future (intentionally, hopefully preventing us from committing changes that break cleanup in some way)

Reply:
node's natural exit is nice to leverage here.

is there a reason we can't really follow the same pattern in this tslint script, as we do in eslint script? it looks like in that one, we throw error and let it bubble up. can we do something like that here? and handle the exit higher?

Reply:
We could if there was a higher-level component, but this is the highest level script for typescript

Reply:
See https://github.com/spalger/kibana/blob/e4e809180de56db64776b0b8767258f79afe19b8/scripts/tslint.js#L21. `scripts/tslint` calls this function directly and it is responsible for exposing the CLI interface

Reply:
Ah, I see. TY!